flyteidl/protos/flyteidl/service/agent.proto

```diff
@@ -8,22 +8,53 @@ import "flyteidl/admin/agent.proto";
 
 // AsyncAgentService defines an RPC Service that allows propeller to send the request to the agent server.
 service AsyncAgentService {
-  // Send a task create request to the agent server.
-  rpc CreateTask (flyteidl.admin.CreateTaskRequest) returns (flyteidl.admin.CreateTaskResponse){};
+  // ExecuteTaskSync streams the create request and inputs to the agent service and streams the outputs back.
+  rpc ExecuteTaskSync (stream flyteidl.admin.ExecuteTaskSyncRequest) returns (stream flyteidl.admin.ExecuteTaskSyncResponse){
+    option (google.api.http) = {
+      post: "/api/v1/agent/task/stream"
+      body: "*"
+    };
+  };
+
+  // CreateTask sends a task create request to the agent service.
+  rpc CreateTask (stream flyteidl.admin.CreateTaskRequest) returns (flyteidl.admin.CreateTaskResponse){
+    option (google.api.http) = {
+      post: "/api/v1/agent/task"
+      body: "*"
+    };
+  };
+
   // Get job status.
-  rpc GetTask (flyteidl.admin.GetTaskRequest) returns (flyteidl.admin.GetTaskResponse){};
+  rpc GetTask (flyteidl.admin.GetTaskRequest) returns (stream flyteidl.admin.GetTaskResponse){
```

Comment:
this stream also seems odd? 
Maybe this is the problem with async vs sync. 
Async being a stream is really odd?